extensions/bitwarden/src/api.ts

```diff
@@ -1,27 +1,71 @@
-import { environment } from "@raycast/api";
+import { environment, getPreferenceValues, LocalStorage, showToast, Toast } from "@raycast/api";
 import { execa, ExecaChildProcess } from "execa";
 import { existsSync } from "fs";
 import { dirname } from "path/posix";
-import { Item, PasswordGeneratorOptions, VaultState } from "./types";
-import { getPasswordGeneratingArgs } from "./utils";
+import { DEFAULT_SERVER_URL } from "./const";
+import { Item, PasswordGeneratorOptions, Preferences, VaultState } from "./types";
+import { getPasswordGeneratingArgs, getServerUrlPreference } from "./utils";
 
 export class Bitwarden {
   private env: Record<string, string>;
   cliPath: string;
-  constructor(clientId: string, clientSecret: string, cliPath: string) {
-    if (!cliPath) {
-      cliPath = process.arch == "arm64" ? "/opt/homebrew/bin/bw" : "/usr/local/bin/bw";
-    }
-    if (!existsSync(cliPath)) {
-      throw new Error(`Bitwarden CLI not found at ${cliPath}`);
+  initPromise: Promise<void>;
```

Comment:
Is this initPromise logic a frequent pattern in javascript ? It took me a bit of time to figure out what was happening.

I would prefer to move the logic to the exec function. I'll propose another solution today.

Reply:
I'm not sure if it is a pattern tbh, thanks for taking the time to review the code.

This was the only way that I could think of that would delay any calls before the server is set. 

For example, the user is logged out and the server is changed. The user enters their password before the status call is complete. Other methods I thought of either a) did the config call twice or b) did it once and let the second go through before it was finished. Some kind of checking on an interval would also work the same as this. 

There's probably a better way, so I look forward to seeing wht you think 👍

Reply:
Thanks for the explanation, I did not thought of the concurrency issue.
Can we move all of logic related to the server to the setServerUrl function ?

Ideally I would like to only have this line in the constructor: 

```
this.initPromise = this.setupServer()
```

with all of the check at the top of the setupServer method, and some early return if nothing has to be done

Reply:
Resolved in e8d428247f168a96a346cf52e5cea704b299ce5e